tools/deploy.go

```diff
@@ -0,0 +1,684 @@
+	}
+	defer outputFile.Close()
+
+	_, err = io.Copy(outputFile, inputFile)
+	if err != nil {
+		return fmt.Errorf("failed to copy from %s to %s: %v", src, dst, err)
+	}
+
+	return nil
+}
+
+func deployBMO(tempBMOOverlay, scriptDir, kustomizePath string) error {
+	revertFunc, err := changeDirTemporarily(tempBMOOverlay)
+	if err != nil {
+		log.Fatalf("Failed to change directory %s: %v", tempBMOOverlay, err)
+	}
+	defer revertFunc()
+
+	// This is to keep the current behavior of using the ironic.env file for the configmap
+	ironicEnvSrc := filepath.Join(scriptDir, "config", "default", "ironic.env")
+	ironicEnvDst := filepath.Join(tempBMOOverlay, "ironic.env")
+	copyFile(ironicEnvSrc, ironicEnvDst)
+
+	execCommand(kustomizePath, "edit", "add", "configmap", "ironic", "--behavior=create", "--from-env-file=ironic.env")
+
+	err = deployWithKustomizeAndApply(kustomizePath, tempBMOOverlay)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func appendToFile(filename, content string) error {
+	f, err := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		log.Fatalf("Failed to open file: %v", err)
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(content)
+	if err != nil {
+		return fmt.Errorf("failed to write to file: %v", err)
+	}
+	return nil
+}
+
+func deployIronic(tempIronicOverlay, scriptDir, kustomizePath string, deployKeepalivedFlag bool) error {
+	revertFunc, err := changeDirTemporarily(tempIronicOverlay)
+	if err != nil {
+		log.Fatalf("Failed to change directory %s: %v", tempIronicOverlay, err)
+	}
+	defer revertFunc()
+
+	// Copy the configmap content from either the keepalived or default kustomization
+	// and edit based on environment.
+	var ironicBMOConfigmapSource string
+	if deployKeepalivedFlag {
+		ironicBMOConfigmapSource = filepath.Join(scriptDir, "ironic-deployment", "components", "keepalived", "ironic_bmo_configmap.env")
+	} else {
+		ironicBMOConfigmapSource = filepath.Join(scriptDir, "ironic-deployment", "default", "ironic_bmo_configmap.env")
+	}
+	ironicBMOConfigmap := filepath.Join(tempIronicOverlay, "ironic_bmo_configmap.env")
+	copyFile(ironicBMOConfigmapSource, ironicBMOConfigmap)
+
+	restartContainerCertificateUpdated := GetEnvOrDefault("RESTART_CONTAINER_CERTIFICATE_UPDATED", "false")
+	cmd := exec.Command("grep", "-q", "RESTART_CONTAINER_CERTIFICATE_UPDATED", ironicBMOConfigmap)
+	err = cmd.Run()
+	if err != nil {
+		exiterr, ok := err.(*exec.ExitError)
+		if ok && exiterr.ExitCode() == 1 {
+			err = appendToFile(ironicBMOConfigmap, fmt.Sprintf("RESTART_CONTAINER_CERTIFICATE_UPDATED=%s\n", restartContainerCertificateUpdated))
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("error checking for RESTART_CONTAINER_CERTIFICATE_UPDATED: %v", err)
+		}
+	} else {
+		execCommand("sh", "-c", fmt.Sprintf("sed -i 's/\\(RESTART_CONTAINER_CERTIFICATE_UPDATED\\).*/\\1=%s/' '%s'", restartContainerCertificateUpdated, ironicBMOConfigmap))
```

Comment:
Not a fan of running external sed for changing a file content. Did not mean to go from a bit flawed `ReplaceAll()` implementation to running commands, especially when these have the exact same issue.

Please revert back to a function that reads in the file, does the replacement (a bit more carefully), and then writes it back.

Reply:
I now changed it to use `strings.Split` to get the exact match in the replacement. I think this will solve the issue you noticed, right?